decimal_test.go

```diff
@@ -95,6 +116,9 @@ func TestNewFromStringErrs(t *testing.T) {
 		".5.2",
 		"8..2",
 		"8.1.",
+		"1e",
+		"1-e",
+		"1e9e",
```

Comment:
Some more test cases you should add:
1ee9
1ee
1e1.2
123.456e1.3
1e-1.2
123.456e-1.3